doc/openssl_migration_guide.rst

```diff
@@ -0,0 +1,389 @@
+        X509* endEntityCert = PEM_read_X509(endEntityCertFileHandle, NULL, NULL, NULL);
+        X509_STORE_CTX_set_cert(ctx, endEntityCert);
+        fclose(endEntityCertFileHandle);
+
+        // Verify the certificate chain
+        int result = X509_verify_cert(ctx);
+        if(result != 1) {
+            // Verification failed
+            X509_STORE_CTX_free(ctx);
+            X509_STORE_free(store);
+            return -1;
+        }
+
+        // Verification succeeded
+        X509_STORE_CTX_free(ctx);
+        X509_STORE_free(store);
+        return 0;
+    }
+
+First, we create a new ``X509_STORE`` object and add the trusted root certificate.
+Then we add the intermediate certificates to the untrusted certificate stack.
+Finally, we set the end-entity certificate and call ``X509_verify_cert()`` to verify
+the whole certificate chain.
+
+Here is the equivalent C++ code using Botan:
+
+.. literalinclude:: /../src/examples/x509_path.cpp
+   :language: cpp
+
+First, we create a ``Certificate_Store_In_Memory`` object and add the trusted root certificate.
+Additionally, we use ``System_Certificate_Store`` to load all trusted root certificates from
+the operating system's certificate store to trust. Botan provides several different
+:ref:`api_ref/x509:certificate stores`, including certificate stores that load certificates
+from a directory or from an SQL database. It even provides an interface for implementing
+your own certificate store.
+Then we add the end-entity certificate and the intermediate certificates to the ``end_certs`` chain.
+Optionally, we can set up path validation restrictions, specify usage and hostname for DNS,
+and then call ``x509_path_validate()`` to :ref:`verify the certificate chain <api_ref/x509:path validation>`.
+
+
+Random Number Generation
+-------------------------
+
+Consider the following application code to generate random bytes using OpenSSL.
+
+.. code-block:: cpp
+
+    #include <openssl/rand.h>
+    #include <iostream>
+
+    int main() {
+        unsigned char buffer[16]; // Buffer to hold 16 random bytes
+
+        if(RAND_bytes(buffer, sizeof(buffer)) != 1) {
+            std::cerr << "Error generating random bytes.\n";
+            return 1;
+        }
+
+        // Print the random bytes in hexadecimal format
+        for(int i = 0; i < sizeof(buffer); i++) {
+            printf("%02X", buffer[i]);
+        }
+        printf("\n");
+
+        return 0;
+    }
+
+This example uses the ``RAND_bytes()`` function to generate 16 random bytes, e.g.,
+for a 128-bit AES key, and prints it on the console.
+
+Here is the equivalent C++ code using Botan:
+
+.. code-block:: cpp
+
+    #include <botan/auto_rng.h>
+    #include <botan/hex.h>
+    #include <iostream>
+
+    int main() {
+        Botan::AutoSeeded_RNG rng;
+
+        Botan::secure_vector<uint8_t> buffer(16); // Buffer to hold 16 random bytes
+        rng.randomize(buffer.data(), buffer.size());
```

Comment:
```suggestion
        const Botan::secure_vector<uint8_t> buffer = rng.random_vec(16);
```

or maybe even:

```suggestion
        const auto buffer = rng.random_vec(16);
```

... but, I think there's some value to spelling out the type for this example.

Alternatively, one can use the more modern `std::span<>` based API and avoid the raw-pointer handling:

```suggestion
        Botan::secure_vector<uint8_t> buffer(16); // Buffer to hold 16 random bytes
        rng.randomize(buffer);
```